app/src/full/java/org/openobservatory/ooniprobe/common/FirebaseMessagingService.java

```diff
@@ -5,32 +5,60 @@
 
 import com.google.firebase.messaging.RemoteMessage;
 
-import org.json.JSONObject;
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.activity.OoniRunActivity;
 
-import java.util.Map;
+import ly.count.android.sdk.messaging.CountlyPush;
 
 public class FirebaseMessagingService extends com.google.firebase.messaging.FirebaseMessagingService {
 	private static final String TAG = "FCM";
 
 	@Override
 	public void onMessageReceived(RemoteMessage remoteMessage) {
-		// TODO(developer): Handle FCM messages here.
-		// Not getting messages here? See why this may be: https://goo.gl/39bRNJ
-		Log.d(TAG, "Message from: " + remoteMessage.getFrom());
-		Log.d(TAG, "Message data payload: " + remoteMessage.getData());
-		Map<String, String> params = remoteMessage.getData();
-		// Check if message contains a data payload.
-		if (remoteMessage.getData().size() > 0) {
-			try {
-				//TODO-FUTURE we can use click_action instead of type
-				JSONObject data = new JSONObject(params.toString());
-			} catch (Exception e) {
-				System.out.println("JSONException " + e);
-			}
+		super.onMessageReceived(remoteMessage);
+
+		Log.d("DemoFirebaseService", "got new message: " + remoteMessage.getData());
+
+		// decode message data and extract meaningful information from it: title, body, badge, etc.
+		CountlyPush.Message message = CountlyPush.decodeMessage(remoteMessage.getData());
+
+		if (message != null && message.has("typ")) {
+			// custom handling only for messages with specific "typ" keys
+			message.recordAction(getApplicationContext());
+			return;
+		}
+
+		Boolean result = CountlyPush.displayMessage(getApplicationContext(), message, R.drawable.notification_icon, null);
+		if (result == null) {
+			Log.i(TAG, "Message wasn't sent from Countly server, so it cannot be handled by Countly SDK");
+		} else if (result) {
+			Log.i(TAG, "Message was handled by Countly SDK");
+		} else {
+			Log.i(TAG, "Message wasn't handled by Countly SDK because API level is too low for Notification support or because currentActivity is null (not enough lifecycle method calls)");
+		}
+	}
+
+	//Handle ooni run JSON. This is not needed for now
+	public void handleJson(CountlyPush.Message message){
+		Intent notificationIntent = null;
+		if (message.has("type") && message.data("type").equals("ooni_run")) {
+			Log.i(TAG, "It's a OONIRun message!");
+			notificationIntent = new Intent(getApplicationContext(), OoniRunActivity.class);
+			notificationIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+			notificationIntent.putExtra("mv",  message.data("mv"));
+			notificationIntent.putExtra("ta",  message.data("ta"));
+			notificationIntent.putExtra("tn",  message.data("tn"));
```

Comment:
I assume the intent is basically not delivered here because of the "not needed" comment, right?